code/modules/admin/verbs/debug.dm

```diff
@@ -29,89 +29,43 @@ Because if you select a player mob as owner it tries to do the proc for
-/client/proc/callproc()
+/client/proc/calladvproc()
 	set category = "Debug"
 	set name = "Advanced ProcCall"
 
 	if(!check_rights(R_DEBUG))
 		return
 
-	spawn(0)
-		var/target = null
-		var/targetselected = 0
-		var/lst[] // List reference
-		lst = new/list() // Make the list
-		var/returnval = null
-		var/class = null
-
-		switch(alert("Proc owned by something?",,"Yes","No"))
-			if("Yes")
-				targetselected = 1
-				class = input("Proc owned by...","Owner",null) as null|anything in list("Obj","Mob","Area or Turf","Client")
-				switch(class)
-					if("Obj")
-						target = input("Enter target:","Target",usr) as obj in world
-					if("Mob")
-						target = input("Enter target:","Target",usr) as mob in world
-					if("Area or Turf")
-						target = input("Enter target:","Target",usr.loc) as area|turf in world
-					if("Client")
-						var/list/keys = list()
-						for(var/client/C)
-							keys += C
-						target = input("Please, select a player!", "Selection", null, null) as null|anything in keys
-					else
-						return
-			if("No")
-				target = null
-				targetselected = 0
-
-		var/procname = input("Proc path, eg: /proc/fake_blood","Path:", null) as text|null
-		if(!procname)
-			return
-
-		// Do not make this a global reference. Global references can be cleared out.
-		if (istype(target, /datum/subsystem/dbcore/))
-			to_chat(usr, "<span class='red'>Never use atom proc call to inject SQL.</span>")
-			message_admins("[key_name(usr)] used atom proc call on the db controller.")
-			log_admin("[key_name(usr)] used atom proc call on the db controller.")
-			return
-
-		if(target && !hascall(target, procname))
-			to_chat(usr, "<span class='red'>Error: callproc(): target has no such call [procname].</span>")
-			return
-
-		var/argnum = input("Number of arguments","Number:",0) as num|null
-		if(!argnum && (argnum!=0))
-			return
-
-		lst.len = argnum // Expand to right length
-		//TODO: make a list to store whether each argument was initialised as null.
-		//Reason: So we can abort the proccall if say, one of our arguments was a mob which no longer exists
-		//this will protect us from a fair few errors ~Carn
-
-		var/i
-		for(i = 1, i < argnum + 1, i++) // Lists indexed from 1 forwards in byond
-			lst[i] = variable_set(src)
+	var/target = null
+	var/class = null
 
-		if(targetselected)
+	switch(alert("Proc owned by something?",,"Yes","No"))
+		if("Yes")
+			class = input("Proc owned by...","Owner",null) as null|anything in list("Obj","Mob","Area or Turf","Client")
+			if(!class)
+				return
+			
+			var/list/stufftocall = list()
+			if(class == "Client")
+				for(var/client/C in clients)
+					stufftocall += C
+			else if(class == "Mob")
+				stufftocall = mob_list
+			else
+				for(var/atom/A in world)
+					if(class == "Obj")
+						if(isobj(A))
+							stufftocall += A
+					else if(class == "Area or Turf")
+						if(isarea(A) || isturf(A))
+							stufftocall += A
+					IN_ROUND_CHECK_TICK //bam. now it doesn't freeze the game anymore
```

Comment:
This is painful (plus you should use |= here anyways). I recommend taking a look at #37103 and using filter_typelist_input() for this instead. Any admin calling an adv proc likely will know at least the parent type of what they are looking for.

Reply:
yea, this was clearly written forever ago (bay era?) and i should change it